src/util/expr.h

```diff
@@ -84,6 +84,9 @@ class exprt:public irept
     return static_cast<const typet &>(find(ID_type));
   }
 
+  /// Amount of nodes this expression tree contains.
+  std::size_t size() const;
```

Comment:
:bulb: This doesn't take into account sharing. Maybe it should, or we could have two versions of it.

Reply:
Sharing shouldn't matter unless a shared tree requires a different way of counting. Whether you have 5 expressions that are all shared or 5 separate, their size should all be considered the same.

Reply:
Perhaps just mention it in the docstring -- `x + x` has `2|x| + 1` nodes, regardless of whether the two expressions are shared or not (i.e. this is the size of the printed tree, not the memory consumption of the expression)